src/tasks/netcore/NetCoreTaskHelper.ts

```diff
@@ -287,6 +298,44 @@ export class NetCoreTaskHelper implements TaskHelper {
 
         return volumes;
     }
+    /**
+     * @see {@link https://learn.microsoft.com/en-us/dotnet/core/docker/publish-as-container} for further information
+     */
+    private async buildWithDotnetSdk(context: DockerBuildTaskContext, buildDefinition: DockerBuildTaskDefinition): Promise<void> {
+        const configuration = buildDefinition.netCore?.configuration ? `-c ${buildDefinition.netCore?.configuration} ` : '';
+        const publishFlag = NetCoreTaskHelper.isWebApp ? '-p:PublishProfile=DefaultContainer' : '/t:PublishContainer';
+        const [runtimeOs, runtimeArch] = this.getOsAndArch(buildDefinition);
+        const imageNameInfo = parseDockerLikeImageName(buildDefinition.dockerBuild.tag);
+        const sdkBuildCommand = `dotnet publish --os ${runtimeOs} --arch ${runtimeArch} ${publishFlag} ${configuration}-p:ContainerImageName=${imageNameInfo.image} -p:ContainerImageTag=${imageNameInfo.tag}`;
+        await context.terminal.execAsyncInTerminal(
+            sdkBuildCommand,
+            {
+                folder: context.folder,
+                token: context.cancellationToken,
+            }
+        );
+    }
+
+    private getOsAndArch(buildDefinition: DockerBuildTaskDefinition): [string, string] {
+        let runtimeOs: string = "linux";
+        let runtimeArch: string = "x64";
+        if (buildDefinition.dockerBuild?.platform) {
+            const [dockerOs, dockerArch] = buildDefinition.dockerBuild.platform.split('/');
+            runtimeOs = dockerOs;

```

Comment:
It looks like we will need to map between Docker and .NET for the OS; it's windows-* for Docker, but win-* for .NET.